great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -124,8 +125,9 @@ class RuleBasedProfiler:
 
     def __init__(
         self,
-        name: str,
-        config_version: float,
+        profiler_config: Optional[RuleBasedProfilerConfig] = None,
```

Comment:
just wanted to check my understanding here: so the `RuleBasedProfiler` has no required parameters to instantiate?

Reply:
shouldn't we at least have a name?

Reply:
@Shinnnyshinshin It does have a name -- it can be supplied explicitly, or as part of the `RuleBasedProfilerConfig` typed object.  Thanks.

Reply:
Ok that makes a lot of sense. Right now, I think the logic only tests for `profiler_config` not being None. Could we add additional checks (and associated error messages) for a situation where 1) both are provided 2) neither are provided?

Reply:
both meaning `profiler_config` and individual parameters

Reply:
@Shinnnyshinshin This should be done outside -- by running it through YAML/config checker.

Reply:
@Shinnnyshinshin constructor full of nulls will raise an error because domain_builder is required in a rule.

Reply:
I see what is going on here; we are allowing both config-based initialization or a "decomposed config initialization". My preference would be to pick one (the config-based one), especially since the only thing calling this constructor is us, and make all the parameters required. Kind of a "why do more when you can do less?" approach.

It might be work revisiting this constructor when we remove DataContext as a dependency, but for now we can tolerate it, assuming there is a good reason to allow for both code paths/methods of initialization